src/AstRunner/AstMap/ClassLikeName.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SensioLabs\Deptrac\AstRunner\AstMap;
+
+final class ClassLikeName
+{
+    /**
+     * @var string
+     */
+    private $className;
+
+    private function __construct(string $className)
+    {
+        $this->className = ltrim($className, '\\');
+    }
+
+    public static function fromString(string $className): self
+    {
+        return new self($className);
+    }
+
+    public function match(string $pattern): bool
+    {
+        return 1 === preg_match($pattern, $this->className);
+    }
+
+    public function __toString()
```

Comment:
If you also want to be a little stricter here, then you should avoid __toString and implement a toString method.

https://github.com/ShittySoft/symfony-live-berlin-2018-doctrine-tutorial/pull/3#issuecomment-460441229